src/js/components/NewConnectionModal.js

```diff
@@ -89,7 +100,16 @@ export default function NewConnectionModal() {
       let form = brim.form(f, config)
 
       if (await form.isValid()) {
-        form.submit()
+        const {host} = form.getFields().reduce((obj, field) => {
```

Comment:
Maybe this reduce call could be in `form.getFields`? It mike make it a little easier to work with.